js/src/api/contract/contract.js

```diff
@@ -269,6 +217,11 @@ export default class Contract {
       }
     }
 
+    return this._subscribe(event, options, callback);
+  }
+
+  _subscribe (event = null, _options, callback) {
+    const subscriptionId = this._subscriptions.length;
```

Comment:
I'm not sure if this approach is 100% safe?
Imagine this:
1. (0) is subscribed
2. (1) is subscribed
3. (0) is unsubscribing
4. (1) is subscribed again

It will work only if subscriptions are unsusbscribing in the same order they were subscribed.